api/src/main/java/com/graphhopper/GHRequest.java

```diff
@@ -243,6 +243,8 @@ public String getWeighting() {
      * By default it supports fastest and shortest. Or specify empty to use default.
      */
     public GHRequest setWeighting(String w) {
+        // todonow: setting weighting/vehicle only works on the client side, but no longer with GHRequest in GraphHopper
+        // should we remove it?
```

Comment:
I wonder if we should remove `get/SetVehicle/Weighting` from `HintsMap` now? The problem is that when using GraphHopper as a Java library we pass `GHRequest` including a `HintsMap` to `GraphHopper#route`, but this method no longer accepts weighting/vehicle but needs a profile instead. On the other hand when we use `GHRequest` on the client side via `GraphHopperWeb` one can still use vehicle/weighting due to the `ProfileResolver` in the `RouteResource` (even though also here profile is the recommended parameter now). We could also  copy&paste `GHRequest` and `HintsMap` on the client side and adjust both versions accordingly, but I do not see a real reason to keep `vehicle/weighting`? When someone is updating the client version there will be a compile error and he/she will be forced to replace `setVehicle/Weighting` with `setProfile`, but is this a problem? Using older clients still works of course (because of `ProfileResolver`).

Reply:
> When someone is updating the client version there will be a compile error and he/she will be forced to replace setVehicle/Weighting with setProfile, but is this a problem?

In general this would be nice to have. Could we even remove HintsMap completely (replace with PMap)?

And I see `hints().putObject("vehicle", "bike")` is still possible so removing this should not be a problem. (E.g. we might not be fast enough to accept "profile" in the public API)

Reply:
> In general this would be nice to have. Could we even remove HintsMap completely (replace with PMap)?

Absolutely I already did this here: https://github.com/graphhopper/graphhopper/commit/d2cc0978c3dd284007816c7de5a1107d90284d7a